### PR TITLE
Add Pairtools sort

### PR DESCRIPTION
**Pull request by diegomics**
FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

### CODE REVIEW

File: tools/pairtools/sort.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            '$pairs_path'
+            -o '$output_sorted_pairs'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="pairs_path" type="data" format="pairs" label="Input pairs file" help="Input .pairs/.pairsam file"/>
+    </inputs>
+    <outputs>
+        <data name="output_sorted_pairs" format="pairs,pairsam" label="${tool.name} on ${on_string}: .pairsam" lines_diff="10"/>
```
Comment (bgruening) [inline tools/pairtools/sort.xml:18]: lines_diff does not work for outputs, only for test outputs

File: tools/pairtools/sort.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        pairtools sort
+            '$pairs_path'
+            -o '$output_sorted_pairs'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="pairs_path" type="data" format="pairs" label="Input pairs file" help="Input .pairs/.pairsam file"/>
```
Comment (bgruening) [inline tools/pairtools/sort.xml:15]: pairs is not a valid Galaxy format

Comment (diegomics) [reply tools/pairtools/sort.xml:15]: :(
in the parse xml we have this line and it worked...

`<data name="output_parsed_pairs" format="pairs"`

I can change it to text,tabular. Also pairsam?

File: tools/pairtools/sort.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            -o '$output_sorted_pairs'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="pairs_path" type="data" format="text,tabular" label="Input pairs file" help="Input .pairs/.pairsam file"/>
+    </inputs>
+    <outputs>
+        <data name="output_sorted_pairs" format="pairs,pairsam" label="${tool.name} on ${on_string}: .pairsam"/>
+        </data>
```
Comment (SaimMomin12) [inline tools/pairtools/sort.xml:19]: Repeated ending tag.

```suggestion
```

**Review by SaimMomin12 — CHANGES_REQUESTED** [review]
Hi @diegomics,

A small suggestion inline.

File: tools/pairtools/sort.xml (hunk 2 above)
Comment (SaimMomin12) [reply tools/pairtools/sort.xml]: I just double checked, .pairs and .pairsam both are tabular format data as per https://pairtools.readthedocs.io/en/latest/formats.html#pairs

If we all agree, we can stick to `"format=tabular"`? If not, we are also happy to include .pairs and .pairsam format to Galaxy.

File: tools/pairtools/sort.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    </outputs>
+    <tests>
+        <!--Test default parameters-->
+        <test expect_num_outputs="1">
+            <param name="pairs_path" value="output_parsed_pairs_bam.pairs"/>
+            <output name="output_sorted_pairs" ftype="pairsam" file="output_sorted_pairs.pairsam" lines_diff="10"/>
```
Comment (SaimMomin12) [inline tools/pairtools/sort.xml:24]: Maybe we can change this to tabular? 
```suggestion
            <output name="output_sorted_pairs" ftype="tabular" file="output_sorted_pairs.pairsam" lines_diff="10"/>
```

File: tools/pairtools/sort.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            '$pairs_path'
+            -o '$output_sorted_pairs'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="pairs_path" type="data" format="text,tabular" label="Input pairs file" help="Input .pairs/.pairsam file"/>
+    </inputs>
+    <outputs>
+        <data name="output_sorted_pairs" format="pairs,pairsam" label="${tool.name} on ${on_string}: .pairsam"/>
```
Comment (SaimMomin12) [inline tools/pairtools/sort.xml]: tabular?

```suggestion
        <data name="output_sorted_pairs" format="tabular" label="${tool.name} on ${on_string}: .pairsam"/>
```

File: tools/pairtools/sort.xml (hunk 2 above)
Comment (diegomics) [reply tools/pairtools/sort.xml]: Ok, I'll update all to tabular

File: tools/pairtools/sort.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        pairtools sort
+            '$pairs_path'
+            -o '$output_sorted_pairs'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="pairs_path" type="data" format="tabular" label="Input pairs file" help="Input .pairs/.pairsam file"/>
```
Comment (SaimMomin12) [inline tools/pairtools/sort.xml:15]:
```suggestion
        <param name="pairs_path" type="data" format="text,tabular" label="Input pairs file" help="Input .pairs/.pairsam file"/>
```

File: tools/pairtools/sort.xml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            <output name="output_sorted_pairs" ftype="tabular" file="output_sorted_pairs.pairsam" lines_diff="10"/>
+        </test>
+
+    </tests>
+    <help><![CDATA[
+        **Pairtools sort**
+
+        Sort pairs in the lexicographic order along chrom1 and chrom2, in the numeric order along pos1 and pos2 and in the lexicographic order along pair_type.
+        
+        pairs_path : Input .pairs or .pairsam file.
```
Comment (bgruening) [inline tools/pairtools/sort.xml:33]:
```suggestion
```

File: tools/pairtools/sort.xml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            '$pairs_path'
+            -o '$output_sorted_pairs'
+            --nproc-in \${GALAXY_SLOTS:-4}
+            --nproc-out \${GALAXY_SLOTS:-4}
+    ]]></command>
+    <inputs>
+        <param name="pairs_path" type="data" format="text,tabular" label="Input pairs file" help="Input .pairs/.pairsam file"/>
+    </inputs>
+    <outputs>
+        <data name="output_sorted_pairs" format="tabular" label="${tool.name} on ${on_string}: .pairsam"/>
```
Comment (bgruening) [inline tools/pairtools/sort.xml:18]:
```suggestion
        <data name="output_sorted_pairs" format="tabular" label="${tool.name} on ${on_string}"/>
```

I guess it makes no sense to specify the datatype again?

**Review by bgruening — COMMENTED** [review]
We need to decide if we add the PAIR datatype to Galaxy or if we go for tabular for the time being.

**Comment by SaimMomin12** [comment]
Resolved: https://github.com/galaxyproject/tools-iuc/pull/5917

**Comment by diegomics** [comment]
Nice! Yes... both generating part of the files for the test dataset messed up the diff. I'm closing this PR now that #5917 is done

**Pull request closed without merging**